processor/blockprocessor/initialize.go

```diff
@@ -26,20 +24,7 @@ import (
 )
 
 // InitializeLedgerSimple executes the migration core functionality.
-func InitializeLedgerSimple(logger *log.Logger, round uint64, opts *idb.IndexerDbOptions) error {
-	ctx, cf := context.WithCancel(context.Background())
-	defer cf()
-	{
-		cancelCh := make(chan os.Signal, 1)
-		signal.Notify(cancelCh, syscall.SIGTERM, syscall.SIGINT)
-		go func() {
-			<-cancelCh
-			logger.Errorf("Ledger migration interrupted")
-			// exit process if migration is interrupted so that
-			// migration state doesn't get updated in db
-			os.Exit(1)
-		}()
-	}
+func InitializeLedgerSimple(ctx context.Context, cf context.CancelFunc, logger *log.Logger, round uint64, opts *idb.IndexerDbOptions) error {
```

Comment:
Is something missing here? How does this exit when it's all caught up?

Reply:
cancel() is called here https://github.com/algorand/indexer/blob/eb2ac877849bf9841e69bd61bd9c5603267bac3a/processor/blockprocessor/initialize.go#L162-L165

Reply:
Thanks, I missed that.

I think this would cause problems because the context is going to be cancelled when we're finished initializing. You can wrap the context with another `WithCancel`. I had to try it, but this seems to work as expected: https://go.dev/play/p/l1S-LqG0MkR

Reply:
fix added.